build.sh

```diff
@@ -197,9 +196,12 @@ function docker_build_with_version {
   echo "$IMAGE_ID" > .image-id
 }
 
-# Versions are stored in subdirectories. You can specify VERSION variable
-# to build just one single version. By default we build all versions
-dirs=${VERSION:-$VERSIONS}
+# This adds backwards compatibility if only single version needs to be built
+# In CI we would like to test single version but VERSIONS= means, that nothing is tested
+# make build TARGET=<OS> VERSIONS=<something> ... checks single version for CLI
+# make build TARGET=<OS> SINGLE_VERSION=<something> ... checks single version from Testing Farm
+dirs=${SINGLE_VERSION:-$VERSIONS}
+echo "Built versions are: $dirs"
```

Comment:
What you want to do here is keep the original behaviour (`dirs=${VERSION:-$VERSIONS}`) only if `SINGLE_VERSION` is not set, otherwise use `SINGLE_VERSION`

Reply:
So something like

```
dirs=${VERSION:-$VERSIONS}
dirs=${SINGLE_VERSION:-$dirs}
```

Reply:
Here is the update. No the same builds as was mentioned in my comment https://github.com/sclorg/container-common-scripts/pull/344/commits/19b3a5abf898cf6d1e7493276778f4773d0d383f

Reply:
> dirs=${SINGLE_VERSION:-$dirs}

@pkubatrh Thank you for the hint, but it does not solve the problem. See:
```bash
make build TARGET=fedora
make[1]: Entering directory '/root/nginx-container'
VERSION="1.20" SKIP_SQUASH=1 OS=fedora CLEAN_AFTER= DOCKER_BUILD_CONTEXT=.. OPENSHIFT_NAMESPACES="" CUSTOM_REPO="" REGISTRY="quay.io/" /usr/bin/env bash common/build.sh
Built versions are: 1.20
-> .exclude-fedora file exists for version 1.20, skipping build.
VERSION="1.22" SKIP_SQUASH=1 OS=fedora CLEAN_AFTER= DOCKER_BUILD_CONTEXT=.. OPENSHIFT_NAMESPACES="" CUSTOM_REPO="" REGISTRY="quay.io/" /usr/bin/env bash common/build.sh
Built versions are: 1.22
-> .exclude-fedora file exists for version 1.22, skipping build.
VERSION="1.22-micro" SKIP_SQUASH=1 OS=fedora CLEAN_AFTER= DOCKER_BUILD_CONTEXT=.. OPENSHIFT_NAMESPACES="" CUSTOM_REPO="" REGISTRY="quay.io/" /usr/bin/env bash common/build.sh
Built versions are: 1.22-micro
-> .exclude-fedora file exists for version 1.22-micro, skipping build.
VERSION="1.24" SKIP_SQUASH=1 OS=fedora CLEAN_AFTER= DOCKER_BUILD_CONTEXT=.. OPENSHIFT_NAMESPACES="" CUSTOM_REPO="" REGISTRY="quay.io/" /usr/bin/env bash common/build.sh
Built versions are: 1.24
-> Version 1.24: building image from 'Dockerfile.fedora' ...
-> Pulling image quay.io/fedora/s2i-core:37 before building image from Dockerfile.fedora.

[snipped]
--> 4926bce99f7c
4926bce99f7c291e215036ab2290cfd42b2f08e9446431af674b35940df0b9f7
nginx 1.24 => 4926bce99f7c291e215036ab2290cfd42b2f08e9446431af674b35940df0b9f7

```
This works properly as before.
With SINGLE_VERSION container  is built forth times:
```bash
make build TARGET=fedora SINGLE_VERSION=1.24
make[1]: Entering directory '/root/nginx-container'
VERSION="1.20" SKIP_SQUASH=1 OS=fedora CLEAN_AFTER= DOCKER_BUILD_CONTEXT=.. OPENSHIFT_NAMESPACES="" CUSTOM_REPO="" REGISTRY="quay.io/" /usr/bin/env bash common/build.sh
Built versions are: 1.24
-> Version 1.24: building image from 'Dockerfile.fedora' ...
-> Pulling image quay.io/fedora/s2i-core:37 before building image from Dockerfile.fedora.

[snipped] - Built for 1st time

--> df867ef4cf33
df867ef4cf3391114fd92a28452a5e2040011690fc7cf250ed4175ea3ba64659
VERSION="1.22" SKIP_SQUASH=1 OS=fedora CLEAN_AFTER= DOCKER_BUILD_CONTEXT=.. OPENSHIFT_NAMESPACES="" CUSTOM_REPO="" REGISTRY="quay.io/" /usr/bin/env bash common/build.sh
Built versions are: 1.24
-> Version 1.24: building image from 'Dockerfile.fedora' ...
-> Pulling image quay.io/fedora/s2i-core:37 before building image from Dockerfile.fedora.
The image quay.io/fedora/s2i-core:37 is already pulled.

[snipped] - Build for 2nd time

--> Using cache df867ef4cf3391114fd92a28452a5e2040011690fc7cf250ed4175ea3ba64659
--> df867ef4cf33
df867ef4cf3391114fd92a28452a5e2040011690fc7cf250ed4175ea3ba64659
VERSION="1.22-micro" SKIP_SQUASH=1 OS=fedora CLEAN_AFTER= DOCKER_BUILD_CONTEXT=.. OPENSHIFT_NAMESPACES="" CUSTOM_REPO="" REGISTRY="quay.io/" /usr/bin/env bash common/build.sh
Built versions are: 1.24
-> Version 1.24: building image from 'Dockerfile.fedora' ...
-> Pulling image quay.io/fedora/s2i-core:37 before building image from Dockerfile.fedora.
The image quay.io/fedora/s2i-core:37 is already pulled.

[snipped] - Built for 3rd time

--> Using cache df867ef4cf3391114fd92a28452a5e2040011690fc7cf250ed4175ea3ba64659
--> df867ef4cf33
df867ef4cf3391114fd92a28452a5e2040011690fc7cf250ed4175ea3ba64659
VERSION="1.24" SKIP_SQUASH=1 OS=fedora CLEAN_AFTER= DOCKER_BUILD_CONTEXT=.. OPENSHIFT_NAMESPACES="" CUSTOM_REPO="" REGISTRY="quay.io/" /usr/bin/env bash common/build.sh
Built versions are: 1.24
-> Version 1.24: building image from 'Dockerfile.fedora' ...
-> Pulling image quay.io/fedora/s2i-core:37 before building image from Dockerfile.fedora.
The image quay.io/fedora/s2i-core:37 is already pulled.

[snipped] - Built for 4th time

--> Using cache df867ef4cf3391114fd92a28452a5e2040011690fc7cf250ed4175ea3ba64659
--> df867ef4cf33
df867ef4cf3391114fd92a28452a5e2040011690fc7cf250ed4175ea3ba64659
nginx 1.24 => df867ef4cf3391114fd92a28452a5e2040011690fc7cf250ed4175ea3ba64659
```

This commit should solve it: https://github.com/sclorg/container-common-scripts/pull/344/commits/19b3a5abf898cf6d1e7493276778f4773d0d383f